File: GameEngine/src/shaders/StaticShader.java
package shaders;

import org.lwjgl.util.vector.Matrix4f;

public class StaticShader extends ShaderProgram {

	
	private static final String VERTEX_FILE = "src/shaders/vertexShader.glsl";
	private static final String FRAGMENT_FILE = "src/shaders/fragmentShader.glsl";
	
	private int transformationMatrixLocation;
	
	
	public StaticShader() {
		
		super(VERTEX_FILE, FRAGMENT_FILE);
	}

	
	@Override
	protected void bindAttributes() {

		super.bindAttribute(0, "position");
		super.bindAttribute(1, "textureCoordinates");
	}


	@Override
	protected void getAllUniformLocations() {
		
		transformationMatrixLocation = super.getUniformLocation("transfomationMatrix");
	}
	
	
	public void injectTransformationMatrix(Matrix4f matrix) {
		super.loadMatrix(transformationMatrixLocation, matrix);
	}

}

File: DungeonHunter/src/shaders/StaticShader.kt
package shaders

import entities.Camera
import utils.Matrix4f
import utils.createViewMatrix


class StaticShader() : ShaderProgram("src/shaders/vertexShader.glsl",
        "src/shaders/fragmentShader.glsl") {

    var locationTransformationMatrix: Int = 0
        private set

    var locationProjectionMatrix: Int = 0
        private set

    var locationViewMatrix: Int = 0
        private set


    override fun bindAttributes() {
        super.bindAttribute(0, "position")
        super.bindAttribute(1, "textureCoords")
    }


    override fun getAllUniformLocations() {
        locationTransformationMatrix = super.getUnifomLocation("transformationMatrix")
        locationProjectionMatrix = super.getUnifomLocation("projectionMatrix")
        locationViewMatrix = super.getUnifomLocation("viewMatrix")
    }


    fun loadTransformationMatrix(matrix: Matrix4f) {
        super.loadMatrix(locationTransformationMatrix, matrix)
    }


    fun loadProjectionMatrix(projection: Matrix4f) {
        super.loadMatrix(locationProjectionMatrix, projection)
    }


    fun loadViewMatrix(camera: Camera) {
        val view = createViewMatrix(camera)
        super.loadMatrix(locationViewMatrix, view)
    }
}
